pkg/cache/cache.go

```diff
@@ -33,25 +33,49 @@ import (
 	"k8s.io/klog/v2"
 )
 
-type CacheResponse struct {
+type Entry struct {
 	RoleARN         string
 	Audience        string
 	UseRegionalSTS  bool
 	TokenExpiration int64
 }
 
+type Request struct {
+	Name                string
+	Namespace           string
+	RequestNotification bool
+}
+
+func (r Request) CacheKey() string {
+	return r.Namespace + "/" + r.Name
+}
+func (r Request) WithNotification() Request {
```

Comment:
The reason I did this can be found here (`handler.go`):
```
	request := cache.Request{Namespace: pod.Namespace, Name: pod.Spec.ServiceAccountName}
	response := m.Cache.Get(request.WithNotification())
	if !response.FoundInSACache && !response.FoundInCMCache && m.saLookupGraceTime > 0 {
		klog.Warningf("Service account %s not found in the cache. Waiting up to %s to be notified", request.CacheKey(), m.saLookupGraceTime)
		select {
		case <-response.Notifier:
			response = m.Cache.Get(request)
			if !response.FoundInSACache && !response.FoundInCMCache {
				klog.Warningf("Service account %s not found in the cache after being notified. Not mutating.", request.CacheKey())
				return nil
			}
		case <-time.After(m.saLookupGraceTime):
			klog.Warningf("Service account %s not found in the cache after %s. Not mutating.", request.CacheKey(), m.saLookupGraceTime)
			return nil
		}
	}
```

The request is created once but without requesting notification. But it is used twice, once with and once without notification. I thought this is cleaner compared to initializing the struct twice with mostly same data.

Reply:
Ah I see. Re-using the same request object across multiple `Cache.Get` calls seems a bit odd, so it might be more declarative to initialize it each time. But I'm fine with either approach, not a blocker.